### PR TITLE
Inform about missing configuration confirmation

### PR DESCRIPTION
**Pull request by rfeistenauer**
## Description

Adds a warning to the readme and config server node to inform the user about the possible mismatch of the loaded config and the one on the connected hardware.

---

<details>
<summary>PR Checklist</summary>

### Things to add, update or check by the maintainers before this PR can be merged.

- [x] ~Public api function documentation~
- [x] ~Architecture documentation reflects actual code structure~
- [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
- [x] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
- [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
- [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [x] ~CHANGELOG.rst updated~
- [x] ~Copyright headers~
- [x] ~Examples~

### Review Checklist
- [x] ~Soft- and hardware architecture (diagrams and description)~
- [x] ~Test review (test plan and individual test cases)~
- [x] Documentation describes purpose of file(s) and responsibilities
- [x] Code (coding rules, style guide)

### Release planning (please answer)
- [x] When is the new feature released?
- [x] ~Which dependent packages do have to be released simultaneously?~

### Hardware tests
_Unstrike the text below to enable automatic hardware tests if available. Otherwise use this as a request for the reviewer._
- [x] ~Perform hardware tests~

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,16 +169,16 @@ The timestamps of the scan data published are computed to be close to reality an
 ### Transferred IOs
 
 The state of inputs and outputs of the PSENscan safety laser scanner are published as 
-([psen_scan_v2/IOState][]). The topic at `/\<name\>/io_states` is latched and always gives 
+[psen_scan_v2/IOState][]. The topic at `/<name>/io_states` is latched and always gives 
 the most recent pin state. On changes at the physical IOs, the changes are logged to
 the console and the new state is published to the topic.
 
-The [input pin state][psen_scan_v2/InputPins] contains information about
+The [input pin states][psen_scan_v2/InputPins] contains information about
```
Comment (agutenkunst) [inline README.md:176]: Please move 269ab4510e020874cd710603aa19626ac02d4c17 into a dedicated PR

File: README.md [hunk 2]
```diff
@@ -200,6 +200,13 @@ You can try this out with:
 roslaunch psen_scan_v2 psen_scan_v2.lauch config_file:='full_path_to/example_config.xml'
 ```
 
+```
+WARNING
+There is no verification this configuration matches the one actually loaded on the PSENscan device!
+
+Please always make sure, to keep those configurations syncronized otherwise this visualization might be missleading!
```
Comment (agutenkunst) [inline README.md:207]: - "those" could be more specific. Like "the configuration you created using the PSENscan configurator"
- Also not only would the visualization be missleading but also the zone polygones could be simply wrong etc

File: src/config_server_node.cpp [hunk 3]
```diff
@@ -32,6 +32,12 @@ ConfigServerNode::ConfigServerNode(ros::NodeHandle& nh, const char* config_file_
     auto zoneconfig = configuration::xml_config_parsing::parseFile(config_file_path);
     zoneset_pub_ = nh_.advertise<::psen_scan_v2::ZoneSetConfiguration>(DEFAULT_ZONESET_TOPIC, 1, true /*latched*/);
 
+    ROS_WARN_STREAM_NAMED(
+        "ConfigurationServer",
+        "The configuration server doesn't verfiy the provided config mathes the one on the connected hardware! "
```
Comment (agutenkunst) [inline src/config_server_node.cpp:37]:
```suggestion
        "The configuration server doesn't verfiy the provided configuration file matches the one on the connected device! "
```

File: src/config_server_node.cpp [hunk 4]
```diff
@@ -32,6 +32,12 @@ ConfigServerNode::ConfigServerNode(ros::NodeHandle& nh, const char* config_file_
...
+        "You are using \"" +
+            std::string(config_file_path) + "\" please verfiy its validity.");
```
Comment (agutenkunst) [inline src/config_server_node.cpp:39]: Maybe also state what could be done to verify the validity.

**Review by agutenkunst — CHANGES_REQUESTED** [review]
See inline comments

File: README.md (hunk 1 above)
Comment (rfeistenauer) [reply README.md:176]: is now in #315

**Review by agutenkunst — APPROVED** [review]
